### PR TITLE
Add support for generation sources into a framework.

### PR DESCRIPTION
**Pull request by thomasvl**
- Add a protoc objc option (generate_for_framework_named) to set the name of
  the framework all generated sources will be in.
- Tweak some comments/naming to make it clear what is the Protobuf framework
  vs. the framework for generated code.
- Update the objc README to document the new generation option to protoc.

This is working towards https://github.com/google/protobuf/issues/1457.

### CODE REVIEW

File: objectivec/README.md [hunk 1]
```diff
@@ -133,6 +133,19 @@ be collisions. This option provides a prefix that will be added to the Enums
...
 
+Objective C Generator `protoc` Options
+--------------------------------------
+
+When generating Objective C code, `protoc` supports a `--objc_opt` argument; the
+argument is comma-delimited name/value pairs (_key=value,key2=value2_). The
+_keys_ are used to change the behavior during generation. The currently
+supported keys are:
+
+  * `generate_for_framework_named`: The `value` used for this key will be used
```
Comment (sergiocampama) [inline objectivec/README.md]: I would rename this to generate_for_named_framework, as it reads a bit better in my opinion

File: src/google/protobuf/compiler/objectivec/objectivec_file.cc [hunk 2]
```diff
@@ -471,7 +496,10 @@ const vector<FileGenerator *> &FileGenerator::DependencyGenerators() {
   return dependency_generators_;
 }
 
-void FileGenerator::PrintFilePreamble(
+// Helper to print the import of the runtime support at the top of generated
+// files. This currently only support the runtime coming from a framework
```
Comment (sergiocampama) [inline src/google/protobuf/compiler/objectivec/objectivec_file.cc]: s/support/supports/

File: objectivec/README.md (hunk 1 above)
Comment (thomasvl) [inline objectivec/README.md]: Done.

File: src/google/protobuf/compiler/objectivec/objectivec_file.cc (hunk 2 above)
Comment (thomasvl) [inline src/google/protobuf/compiler/objectivec/objectivec_file.cc]: Done

**Pull request merged**